plugin-server/tests/worker/ingestion/cookielessServerHashStep.test.ts

```diff
@@ -0,0 +1,371 @@
+
+                if (!actual?.properties) {
+                    throw new Error('no event or properties')
+                }
+                expect(actual.distinct_id).not.toEqual(COOKIELESS_SENTINEL_VALUE)
+                expect(actual.distinct_id.startsWith('cookieless_')).toBe(true)
+                expect(actual.properties.$session_id).toBeDefined()
+            })
+            it('should give the same session id and distinct id to events with the same hash properties and within the same day and session timeout period', async () => {
+                const [actual1] = await cookielessServerHashStep(hub, event)
+                const [actual2] = await cookielessServerHashStep(hub, eventABitLater)
+
+                if (!actual1?.properties || !actual2?.properties) {
+                    throw new Error('no event or properties')
+                }
+                expect(actual2.distinct_id).toEqual(actual1.distinct_id)
+                expect(actual1.properties.$session_id).toBeDefined()
+                expect(actual2.properties.$session_id).toEqual(actual1.properties.$session_id)
+            })
+            it('should give different distinct id and session id to a user with a different IP', async () => {
+                const [actual1] = await cookielessServerHashStep(hub, event)
+                const [actual2] = await cookielessServerHashStep(hub, eventOtherUser)
+                if (!actual1?.properties || !actual2?.properties) {
+                    throw new Error('no event or properties')
+                }
+                expect(actual1.distinct_id).not.toEqual(actual2.distinct_id)
+                expect(actual1.properties.$session_id).not.toEqual(actual2.properties.$session_id)
+            })
+            it('should give different distinct id and session id to events on different days', async () => {
+                const [actual1] = await cookielessServerHashStep(hub, event)
+                jest.setSystemTime(differentDay) // advance time to the next day
+                const [actual2] = await cookielessServerHashStep(hub, eventDifferentDay)
+                if (!actual1?.properties || !actual2?.properties) {
+                    throw new Error('no event or properties')
+                }
+                expect(actual1.distinct_id).not.toEqual(actual2.distinct_id)
+                expect(actual1.properties.$session_id).not.toEqual(actual2.properties.$session_id)
+            })
+            it('should strip the PII used in the hash', async () => {
+                const [actual] = await cookielessServerHashStep(hub, event)
+
+                if (!actual?.properties) {
+                    throw new Error('no event or properties')
+                }
+                expect(actual.ip).toBeNull()
+                expect(actual.properties.$raw_user_user).toBeUndefined()
+                expect(actual.properties.$ip).toBeUndefined()
+                expect(actual.properties.$cookieless_extra).toBeUndefined()
```

Comment:
You're not adding anything to the `$cookieless_extra` property, do you mind doing that? This is not as much for the code we have now, but for future changes

Reply:
Done